airflow/models/skipmixin.py

```diff
@@ -175,6 +206,7 @@ def skip_all_except(
         branch_task_ids is stored to XCom so that NotPreviouslySkippedDep knows skipped tasks or
         newly added tasks should be skipped when they are cleared.
         """
+        log = cls().log  # Note: need to catch logger form instance, static logger breaks pytest
```

Comment:
```suggestion
        log = cls.logger 
```

Reply:
Oh, no. Needed to revert the proposal. I am not sure why but somehow pytests fail. I tried all possible combinations:
1) cls.logger() - generates a KeyError like:
```
self = <Task(ShortCircuitOperator): short_circuit>

    def __getstate__(self):
        state = dict(self.__dict__)
        if self._log:
>           del state["_log"]
E           KeyError: '_log'

airflow/models/baseoperator.py:1393: KeyError
```

2) using the calling class logger
3) Instantiating a different separate logger

It might be only an artifact of pytests that I do not understand - still OK to merge with the note in the code?

Reply:
That's really weird ... I am not a big fan of `cls().log`, I am not sure what does it do behind the scene. If you feel comfortable, feel free to merge